notifier/webhook/deliverer_test.go

```diff
@@ -0,0 +1,163 @@
+package webhook
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/quay/clair/v4/notifier"
+	"github.com/quay/clair/v4/notifier/keymanager"
+	"github.com/quay/claircore/test/log"
+	"gopkg.in/square/go-jose.v2"
+)
+
+var (
+	callback = "http://clair-notifier/api/v1/notification"
+	noteID   = uuid.New()
+)
+
+// TestDeliverer is a parallel test harness
+func TestDeliverer(t *testing.T) {
+	t.Run("TestSign", testSign)
+	t.Run("TestDeliverer", testDeliverer)
+}
+
+// testSign confirms the deliverer correctly signs a webhook
+func testSign(t *testing.T) {
+	t.Parallel()
+
+	d := Deliverer{}
+	kp := (genKeyPair(t, 1))[0]
+	req, err := http.NewRequest(http.MethodGet, "", nil)
+	if err != nil {
+		t.Fatalf("failed to create request: %v", err)
+	}
+
+	ctx := log.TestLogger(context.Background(), t)
+	err = d.sign(ctx, req, kp)
+	if err != nil {
+		t.Fatalf("failed to sign request: %v", err)
+	}
+
+	kid := req.Header.Get("kid")
```

Comment:
The spec says nothing about this HTTP header

Reply:
fixed